### DISCUSSION TITLE
Claude API failing on lookup, unsure why.

### DISCUSSION THREAD

**Discussion by InusahLore42** [brianpetro/obsidian-smart-connections#929] [General]
Im attempting to make use of smart connections for obsidian but it seems like the lookup function fails when using Claude 3.5. - Sonnet.

These are the most recent logs I have, any insight on how to resolve this issue? 

plugin:smart-connections:13179 request_params {url: 'https://api.anthropic.com/v1/messages', method: 'POST', headers: {…}, body: '{"model":"claude-3-5-sonnet-latest","max_tokens":4…}],"tool_choice":{"type":"tool","name":"lookup"}}'}
plugin:smart-connections:13182 http_resp SmartHttpObsidianResponseAdapter {response: {…}}response: {status: 200, headers: {…}, arrayBuffer: ArrayBuffer(589)}[[Prototype]]: SmartHttpResponseAdapter3
plugin:smart-connections:17829 filter {limit: '10'}[[Prototype]]: Object
plugin:smart-connections:9040 Fetch finished loading: POST "https://api.openai.com/v1/embeddings".
request @ plugin:smart-connections:9040
request @ plugin:smart-connections:9005
request @ plugin:smart-connections:9197
embed_batch @ plugin:smart-connections:9145
await in embed_batch (async)
invoke_adapter_method @ plugin:smart-connections:8526
embed_batch @ plugin:smart-connections:8704
lookup @ plugin:smart-connections:6040
handle_lookup_tool_call @ plugin:smart-connections:17759
handle_tool_calls @ plugin:smart-connections:17735
handle_message_from_chat_model @ plugin:smart-connections:17332
await in handle_message_from_chat_model (async)
complete @ plugin:smart-connections:17402
await in complete (async)
init @ plugin:smart-connections:18244
await in init (async)
eval @ plugin:smart-connections:4997
create_or_update @ plugin:smart-connections:4996
handle_message_from_user @ plugin:smart-connections:17283
eval @ plugin:smart-connections:16994
plugin:smart-connections:6066 Found and returned 10 smart_blocks.
plugin:smart-connections:8121 Saving smart_threads: 1 items
plugin:smart-connections:8134 Saved smart_threads in 5ms
plugin:smart-connections:17568 Moved last user message to the end of the request for better context handling.
plugin:smart-connections:13206 request_params {url: 'https://api.anthropic.com/v1/messages', method: 'POST', headers: {…}, body: '{"model":"claude-3-5-sonnet-latest","max_tokens":4…type":"text","text":"tell me about my notes"}]}]}'}
plugin:smart-connections:12740     POST https://api.anthropic.com/v1/messages 400
stream @ plugin:smart-connections:12740
eval @ plugin:smart-connections:13239
stream @ plugin:smart-connections:13207
invoke_adapter_method @ plugin:smart-connections:12337
stream @ plugin:smart-connections:12487
complete @ plugin:smart-connections:17391
await in complete (async)
eval @ plugin:smart-connections:18063
plugin:smart-connections:13234 Stream error: CustomEvent {isTrusted: false, data: '{"type":"error","error":{"type":"invalid_request_e…se` or `tool_result` blocks must define tools."}}', source: SmartStreamer, detail: null, type: 'error', …}
eval @ plugin:smart-connections:13234
eval @ plugin:smart-connections:12717
dispatchEvent @ plugin:smart-connections:12716
#onStreamFailure @ plugin:smart-connections:12762
#onStreamProgress @ plugin:smart-connections:12772
XMLHttpRequest.send (async)
stream @ plugin:smart-connections:12740
eval @ plugin:smart-connections:13239
stream @ plugin:smart-connections:13207
invoke_adapter_method @ plugin:smart-connections:12337
stream @ plugin:smart-connections:12487
complete @ plugin:smart-connections:17391
await in complete (async)
eval @ plugin:smart-connections:18063
plugin:smart-connections:17432 error_handler *API Error. See console logs for details.*
error_handler @ plugin:smart-connections:17432
eval @ plugin:smart-connections:13235
eval @ plugin:smart-connections:12717
dispatchEvent @ plugin:smart-connections:12716
#onStreamFailure @ plugin:smart-connections:12762
#onStreamProgress @ plugin:smart-connections:12772
XMLHttpRequest.send (async)
stream @ plugin:smart-connections:12740
eval @ plugin:smart-connections:13239
stream @ plugin:smart-connections:13207
invoke_adapter_method @ plugin:smart-connections:12337
stream @ plugin:smart-connections:12487
complete @ plugin:smart-connections:17391
await in complete (async)
eval @ plugin:smart-connections:18063
plugin:smart-connections:18064 Uncaught (in promise) CustomEvent {isTrusted: false, data: '{"type":"error","error":{"type":"invalid_request_e…se` or `tool_result` blocks must define tools."}}', source: SmartStreamer, detail: null, type: 'error', …}
eval @ plugin:smart-connections:18064
plugin:smart-connections:12740 XHR failed loading: POST "https://api.anthropic.com/v1/messages".
stream @ plugin:smart-connections:12740
eval @ plugin:smart-connections:13239
stream @ plugin:smart-connections:13207
invoke_adapter_method @ plugin:smart-connections:12337
stream @ plugin:smart-connections:12487
complete @ plugin:smart-connections:17391
await in complete (async)
eval @ plugin:smart-connections:18063

**Comment by brianpetro** [comment]
Hi @InusahLore42 and thanks for the report

In the settings (gear icon in the top-right corner of the chat) do you have the "Send Tool Output in User Message" setting toggled on?

It seems like that setting, when turned on, is causing the above error with anthropic.

![Screenshot 2024-12-17 at 9 13 30 PM](https://github.com/user-attachments/assets/90626466-848d-4d2f-a3a9-3f00e164ef06)

While it still needs to be fixed, turning off that setting should fix the issue in the meantime.

Let me know if that works 🌴

**Reply by InusahLore42** [reply]
That worked, thanks!

**Reply by brianpetro** [reply]
@InusahLore42 should be fixed in the latest version `2.3.46` 🌴
